### DISCUSSION TITLE
partial integro-differential equation

### DISCUSSION THREAD

**Discussion by xyysharon** [zwicker-group/py-pde#189] [Q&A] [answered]
Dear Dr. Zwicker,
Thanks for this wonderful package! I really appreciate your work, but I just have some questions.
I am trying to solve a partial integro-differential equation in 1D which looks like:
 
![image](https://user-images.githubusercontent.com/57993404/146330965-2085d07f-db60-459c-927f-c2e0a9882195.png)

How can I include the integral term in the PDE? I could not find example codes that include integral term.
Can you show demonstration code?
Thank you!

**Comment by david-zwicker** [comment]
This is not supported out-of-the box, but you could write a custom pde class, where you calculate the integral in each timestep.

**Comment by david-zwicker** [comment, marked as answer]
I just realized that it will be rather simple to support integrals over entire fields in the `pde.PDE` class. I started to work on this today and if nothing fails, this should be merged into the main branch tonight.

**Reply by xyysharon** [reply]
Thank you for your response! I'm really looking forward to it！

**Reply by david-zwicker** [reply]
The master branch now supports the following syntax:

```
eq = pde.PDE({'c': '-integral(c)'})
```

This is of course an oversimplified example, where the integral of the field `c` affects the evolution rate at each point. However, this should get you started to implement your own equation.

**Reply by xyysharon** [reply]
Thanks David! It works fine for my work!!! By the way, when I use matplotlib 3.5 for drawing, there was a flash back. So I switched to version 3.2 and it works.

**Reply by david-zwicker** [reply]
Good to hear that things work for you. Concerning the newer matplotlib, could you please open an Issue with a detailed description, so we can track and fix this problem?!

**Reply by xyysharon** [reply]
Okay, I'll open a new one.
